service_container/tags.rst

```diff
@@ -564,3 +564,242 @@ application handlers.
+``index_attribute_name``. As it's an iterator, we can use ``iterator_to_array``
+to get an array and then retrieve the ``handler_two`` handler :
+
+.. code-block:: php
+
+    // src/Handler/HandlerCollection.php
+    namespace App\Handler;
+
+    class HandlerCollection
+    {
+        public function __construct(iterable $handlers)
+        {
+            $handlers = iterator_to_array($handlers);
+
+            $handler_two = $handlers['handler_two']:
+            $handler_one = $handlers['handler_one'];
+        }
+    }
+
+
+.. tip::
+
+    You can omit the ``index_attribute_name`` attribute, by implementing a static
+    method named like this : ``getDefaultIndexAttributeNameName``
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/services.yaml
+            services:
+                App\Handler\One:
+                    tags:
+                        - { name: 'app.handler', priority: 20 }
+
+                App\Handler\Two:
+                    tags:
+                        - { name: 'app.handler', index: 'handler_two' }
+
+                App\HandlerCollection:
+                    # inject all services tagged with app.handler as first argument
+                    arguments: [!tagged { tag: 'app.handler', index_by: 'index' }]
+
+        .. code-block:: xml
+
+            <!-- config/services.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+                <services>
+                    <service id="App\Handler\One">
+                        <tag name="app.handler" priority="20" />
+                    </service>
+
+                    <service id="App\Handler\Two">
+                        <tag name="app.handler" index="handler_two" />
+                    </service>
+
+                    <service id="App\HandlerCollection">
+                        <!-- inject all services tagged with app.handler as first argument -->
+                        <argument type="tagged" tag="app.handler" index-by="index" />
+                    </service>
+                </services>
+            </container>
+
+        .. code-block:: php
+
+            // config/services.php
+            $container->register(App\Handler\One::class)
+                ->addTag('app.handler', ['priority' => 20]);
+
+            $container->register(App\Handler\One::class, ['index' => 'handler_two']);
+
+            $container->register(App\HandlerCollection::class)
+                // inject all services tagged with app.handler as first argument
+                ->addArgument(new TaggedIteratorArgument('app.handler', 'index'));
+
+    And here is how your ``App\Handler\One`` should now be :
```

Comment:
It might be more clear if you show this FIRST, and THEN the changes to `services.yaml` thanks to this (being able to omit the index key)